### DISCUSSION TITLE
Obsidian Release v1.1.15 - plugin  stopped working

### DISCUSSION THREAD

**Discussion by beresarus** [ObsidianToAnki/Obsidian_to_Anki#410] [General]
Unfortunately, after updating Obsidian to version 1.1.15, the Obsidian_to_Anki plugin stopped working. After returning to version 1.1.9. the plugin started working fine. 
Now, for the plugin to work successfully, you will have to forget about updating Obsidian above version 1.1.9.
It's a pity that the author abandoned the plugin update....

**Comment by beresarus** [comment]
I also wrote on the [Obsidian forum](https://forum.obsidian.md/t/obsidian-release-v1-1-15-some-plugins-stopped-working/55079). The following response came:

> Contact the author of that plugins. we don’t take bug reports involving third party plugins. If the problem is within Obsidian, the plugin dev should contact us.

**Reply by HuTuJun** [reply]
my English is not good. I hope you can understand it. 

You can modify this method in the plug-in。

path  ：  `.obsidian\plugins\obsidian-to-anki-plugin\main.js  -> getFolderPathList(file) `

Original content：

``` js
while (abstractFile.hasOwnProperty('parent')) {
```

Modify to:

``` js
    getFolderPathList(file) {
        let result = [];
        let abstractFile = file;
        while (abstractFile != null && abstractFile.hasOwnProperty('parent')) {
            result.push(abstractFile.parent);
            abstractFile = abstractFile.parent;
        }
        result.pop(); // Removes top-level vault
        return result;
    }
```

Restart the application and it will succeed. At least I can。



这个写给我同胞的。
情况看起来应该是获取父级为null。导致报错，后面就不执行了，
可以修改插件里面的 main.js 中的方法  getFolderPathList(file) ， 
新增加一个 null 判断就可以， 忽略上面机翻的英语。改代码就可以了

**Reply by Fred-209** [reply]
This worked for me as well!
Thank you.

**Reply by weisbert** [reply]
It works!! Thank you very very much.

**Comment by giatekla** [comment]
I noticed this too. time to switch over to another plug-in, oof...

**Reply by beresarus** [reply]
**giatekla**, this plugin is great. And it is unlikely that you will find a better one.
We need to find a developer who will fix all the errors and create a fork for ObsidianToAnki.
Who has any ideas? Maybe write on the forum https://forums.ankiweb.net / ?
